### ISSUE TITLE
feat: add the ability to finish connection and channel handshakes for a specified connection/channel

### ISSUE THREAD

**Issue by jtieri** [cosmos/relayer#1127]
Connection and Channel initialization in IBC is a 4 step handshake process involving an `Init` `Try` `Ack` and `Confirm` step. It is possible to experience a failure at any step in the process which leaves the handshake process unfinished. Currently in the relayer we assume the handshake is always starting at the `Init` step but as a user it is desirable to have a way to finish a partially completed handshake. 

Ex: I start to open a connection between two IBC enabled chains. Due to network conditions my attempts to initialize the connection fail and the connection is stuck at `Try`, I don't want to create a whole new connection but would rather finish the handshake for this particular connection. I would like to specify the connection identifier and the current connection state i.e. `Try` so that the relayer will pick the handshake up from there and finish it.

One way to do this would be to accept an optional conn/channel identifier and if present the relayer will query the connection or channel state and then finish the handshake process. This would remove the need for the user to know which state the connection or channel is stuck in so they would just pass in the identifier and the relayer would figure the rest out.

**Comment by jonathanpberger** [comment]
Good idea, but not feasible atm; related to this PR: https://github.com/cosmos/relayer/pull/1147. Iceboxing for now.

**Comment by migueldingli1997** [comment]
This would be very helpful.
